pkg/gohai/filesystem/filesystem_nix.go

```diff
@@ -0,0 +1,196 @@
+// This file is licensed under the MIT License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2014-present Datadog, Inc.
+
+//go:build darwin || linux
+// +build darwin linux
+
+package filesystem
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/sys/unix"
+
+	log "github.com/cihub/seelog"
+	"github.com/moby/sys/mountinfo"
+)
+
+// These FS types should be excluded from listing
+var IgnoredFSTypes = map[string]struct{}{
+	"autofs":      {},
+	"debugfs":     {},
+	"devfs":       {},
+	"devpts":      {},
+	"devtmpfs":    {},
+	"fuse.portal": {},
+	"fusectl":     {},
+	"ignore":      {},
+	"kernfs":      {},
+	"mqueue":      {},
+	"none":        {},
+	"proc":        {},
+	"rpc_pipefs":  {},
+	"squashfs":    {},
+	"subfs":       {},
+	"sysfs":       {},
+}
+
+// These FS types are known to be remote
+var RemoteFSTypes = map[string]struct{}{
+	"acfs":       {},
+	"afs":        {},
+	"auristorfs": {},
+	"coda":       {},
+	"fhgfs":      {},
+	"gpfs":       {},
+	"ibrix":      {},
+	"ocfs2":      {},
+	"vxfs":       {},
+}
+
+// FSInfoGetter provides function to get information about a given filesystem: its size and its dev id
+type FSInfoGetter interface {
+	// Size returns the size of the given filesystem
+	Size(mount *mountinfo.Info) (uint64, error)
+
+	// Dev returns the dev id of the given filesystem
+	// the return type is interface{} because `syscall.Stat_t` uses different types for Dev depending on the platform
+	Dev(mount *mountinfo.Info) (interface{}, error)
+}
```

Comment:
OK, this seems to be used in the test if I'm correct. I don't think we should export this outside the package. Also when making the code easier to test we should aim at the simplest solution. I think an interface with it's different implementation are a bit complex to mock 2 functions.

WDYT about:
```
// wrappers around function to ease testing
var (
	getSize = mountInfoSize
	getDev = mountInfoDev
)

func mountInfoSize(mount *mountinfo.Info) (uint64, error) {
	[...]
}

func mountInfoDev(mount *mountinfo.Info) (uint64, error) {
	[...]
}
```

And in the tests:
```
func TestSomething(t *testing.T) {
	getSize := func() { [...] }
	getDev := func(){ [...] }
	t.Cleanup(func(){
		getSize = mountInfoSize
		getDev = mountInfoDev
	})
}
```

Reply:
I already have 5-6 bugs where the global states for testing are not restored correctly.

Reply:
I think having a whole interface with 2 implementations for a single test is very heavy and add a lot of code for nothing (code that we have to maintain). It adds a lot of complexity for two 5 lines functions.

We can leverage `t.Cleanup()` to have restore the previous state. If you want to avoid this, then simply take to function as parameter to avoid the multiple interfaces. `Size`  and `Dev` are 5 lines each, no need to complexify things for this, in my opinion.

Reply:
I went with providing those functions as parameter in 89d38950be5db6dd400681942b8042625b329e7d, which seems to me like a good middle ground (in particular it removes the interface and test helper functions)